schemas/saveObservations.js

```diff
@@ -9,12 +9,21 @@ const saveObservationsSchema = {
     observerReportTxId: {
       type: 'string',
       pattern: '^[a-zA-Z0-9-_]{43}$',
+      description: 'The transaction ID of the submitted report',
+    },
+    gatewayAddress: {
+      type: 'string',
+      pattern: '^[a-zA-Z0-9-_]{43}$',
+      description:
+        'The gateway address that sent the observation. If not provided, the caller address will be used.',
     },
     failedGateways: {
       type: 'array',
       items: {
         type: 'string',
-        pattern: '^[a-zA-Z0-9-_]{43}$',
+        pattern: '^(?:(?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{1,63}$', // eslint-disable-line no-useless-escape
```

Comment:
observers submit reports for failed `fqdns` atm and we will start to enforce uniqueness of those